src/zenon.cpp

```diff
@@ -389,41 +397,80 @@ void zenon::create_cmd_list()
     kernel_names.clear();
     if (!resnet)
     {
-        uint32_t number_of_kernels = 40;
+        if (disable_blitter) {
+            uint32_t number_of_kernels = 40;
 
-        for (int i = 0; i < number_of_kernels + 2; i++)
-        {
-            zeCommandListAppendEventReset(command_list, kernel_ts_event[i]);
-        }
+            for (int i = 0; i < number_of_kernels + 2; i++)
+            {
+                zeCommandListAppendEventReset(command_list, kernel_ts_event[i]);
+            }
 
-        submit_kernel_to_cmd_list(add_buffers_kernel, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[0], { nullptr }, 0);
-        submit_kernel_to_cmd_list(add_buffers_kernel, { input1_buffer, input2_buffer }, im_buf2, kernel_ts_event[1], { nullptr }, 0);
-        submit_kernel_to_cmd_list(add_buffers_kernel, { input1_buffer, input2_buffer }, im_buf3, kernel_ts_event[2], { nullptr }, 0);
+            submit_kernel_to_cmd_list(set_n_to_output, { input1_buffer, input2_buffer }, im_buf1, kernel_ts_event[0], { nullptr }, 0, 1);
+            submit_kernel_to_cmd_list(set_n_to_output, { input1_buffer, input2_buffer }, im_buf2, kernel_ts_event[1], { nullptr }, 0, 2);
+            submit_kernel_to_cmd_list(set_n_to_output, { input1_buffer, input2_buffer }, im_buf3, kernel_ts_event[2], { nullptr }, 0, 3);
 
-        
 
-        for (int i = 1; i < number_of_kernels; i++)
-        {
-            if (i % 3 == 0)
-                submit_kernel_to_cmd_list(add_buffers_kernel, { im_buf1, im_buf2 }, im_buf3, kernel_ts_event[i + 2], { &kernel_ts_event[i] , &kernel_ts_event[i + 1] }, 2);
-            if (i % 3 == 1)
-                submit_kernel_to_cmd_list(add_buffers_kernel, { im_buf3, im_buf2 }, im_buf1, kernel_ts_event[i + 2], { &kernel_ts_event[i] , &kernel_ts_event[i + 1] }, 2);
-            if (i % 3 == 2)
-                submit_kernel_to_cmd_list(add_buffers_kernel, { im_buf1, im_buf3 }, im_buf2, kernel_ts_event[i + 2], { &kernel_ts_event[i] , &kernel_ts_event[i + 1] }, 2);
+
```

Comment:
code duplication here is huge. please find a way to split only that parts that really differ